src/apps/external_executor/executor_code_id.h

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "ccf/ds/json.h"
+#include "ccf/ds/quote_info.h"
+#include "ccf/service/code_digest.h"
+#include "ccf/service/map.h"
+#include "ccf/service/tables/code_id.h"
+
+struct GetExecutorCode
+{
+  struct Version
+  {
+    std::string digest;
+    ccf::CodeStatus status;
+    std::optional<ccf::QuoteFormat> platform;
+  };
+
+  struct Out
+  {
+    std::vector<GetExecutorCode::Version> versions = {};
+  };
+};
+
+DECLARE_JSON_TYPE_WITH_OPTIONAL_FIELDS(GetExecutorCode::Version)
+DECLARE_JSON_REQUIRED_FIELDS(GetExecutorCode::Version, digest, status)
+DECLARE_JSON_OPTIONAL_FIELDS(GetExecutorCode::Version, platform)
+DECLARE_JSON_TYPE(GetExecutorCode::Out)
+DECLARE_JSON_REQUIRED_FIELDS(GetExecutorCode::Out, versions)
+
+struct ExecutorCodeInfo
+{
+  ccf::CodeStatus status;
+  ccf::QuoteFormat platform;
+};
+
+DECLARE_JSON_TYPE(ExecutorCodeInfo);
+DECLARE_JSON_REQUIRED_FIELDS(ExecutorCodeInfo, status, platform);
+
+using ExecutorCodeIDs = ccf::ServiceMap<ccf::CodeDigest, ExecutorCodeInfo>;
+namespace Tables
```

Comment:
This should be `namespace tables` (lowercase), and the namespace could be removed entirely.

The existing `ccf::Tables` namespace breaks this convention (namespace names are lowercase), because it was originally a single class with a bunch of static members, and we wanted to avoid a breaking change when that was sharded into a namespace.

Reply:
Ah okay. Yeah I did think of removing the namespace itself, done now. Thanks.